cli/module_build.go

```diff
@@ -458,14 +470,53 @@ func (c *viamClient) waitForBuildToFinish(buildID, platform string) (map[string]
 				if platform == "" || job.Platform == platform {
 					status := jobStatusFromProto(job.Status)
 					statuses[job.Platform] = status
+
+					// Handle progress spinner for build steps
+					if pm != nil && job.BuildStep != nil && job.GetBuildStep() != "" && job.GetBuildStep() != lastBuildStep {
```

Comment:
(minor) I don't think we need `job.BuildStep != nil` here, since `job.GetBuildStep()` returns `""` if `job.BuildStep == nil`